lib/oban/plugins/cron.ex

```diff
@@ -170,9 +170,17 @@ defmodule Oban.Plugins.Cron do
   defp valid_crontab?({expression, worker, opts}) do
     %Expression{} = Expression.parse!(expression)
 
-    Code.ensure_loaded?(worker) and
-      function_exported?(worker, :perform, 1) and
-      Keyword.keyword?(opts)
+    !Code.ensure_loaded?(worker) and
+      raise ArgumentError, "#{inspect(worker)} worker is not found or can't be loaded"
+
+    !function_exported?(worker, :perform, 1) and
+      raise ArgumentError, "#{inspect(worker)} worker does not implement `perform/1` callback"
+
+    !Keyword.keyword?(opts) and
+      raise ArgumentError,
+            "crontab options to #{inspect(worker)} worker should be provided as a keyword list"
```

Comment:
Now that all of the checks will raise we can change this from `valid_crontab?` to `validate_crontab!`. Here I've switched to use `unless` and tweaked the wording a bit:

```suggestion
    unless Code.ensure_loaded?(worker) do
      raise ArgumentError, "#{inspect(worker)} not found or can't be loaded"
    end

    unless function_exported?(worker, :perform, 1) do
      raise ArgumentError, "#{inspect(worker)} does not implement `perform/1` callback"
    end

    unless Keyword.keyword?(opts) do
      raise ArgumentError, "options for #{inspect(worker)} must be as a keyword list"
    end
```